util/hash.h

```diff
@@ -49,12 +49,12 @@ struct SliceHasher {
   uint32_t operator()(const Slice& s) const { return GetSliceHash(s); }
 };
 
-// An alternative to % for mapping hashing to an arbitrary range. See
+// An alternative to % for mapping a hash value to an arbitrary range. See
 // https://github.com/lemire/fastrange and
 // https://github.com/pdillinger/wormhashing/blob/2c4035a4462194bf15f3e9fc180c27c513335225/bloom_simulation_tests/foo.cc#L57
 inline uint32_t fastrange32(uint32_t a, uint32_t h) {
-  uint64_t product = (uint64_t)a * h;
-  return (uint32_t)(product >> 32);
+  uint64_t product = static_cast<uint64_t>(a) * h;
```

Comment:
I'm surprised I don't see any compiler to complain while h stays uint32_t. I expect that some compilers (with out setting) will require us to cast h to uint64_t too.

Reply:
> Awesome! Please introduce this improvement in HISTORY.md.

See https://github.com/facebook/rocksdb/pull/5762/commits/a413d854cfe5db695bf0829f43f38f94801aec41